src/VisualStudio/Core/Def/Implementation/Remote/JsonRpcClient.cs

```diff
@@ -59,10 +59,9 @@ protected virtual void Disconnected(JsonRpcDisconnectedEventArgs e)
             }
             catch (Exception ex) when (ReportUnlessCanceled(ex, cancellationToken))
             {
-                // any exception can be thrown from StreamJsonRpc if JsonRpc is disposed in the middle of read/write.
-                // until we move to newly added cancellation support in JsonRpc, we will catch exception and translate to
-                // cancellation exception here. if any exception is thrown unrelated to cancellation, then we will rethrow
-                // the exception
+                // StreamJsonRpc throws RemoteInvocationException if the call is cancelled.
+                // Handle this case by throwing a proper cancellation exception instead.
+                // See https://github.com/Microsoft/vs-streamjsonrpc/issues/67
                 cancellationToken.ThrowIfCancellationRequested();
 
                 LogError($"exception: {ex.ToString()}");
```

Comment:
(having to comment here since GitHub doesn't let me comment outside the diff...)

Should the comment on line 69 ("this is to make us not crash...") be revised with a bit more detail? I admit my biggest confusion before was trying to decipher what was going on there.